internal/pkg/exporter/health.go

```diff
@@ -0,0 +1,77 @@
+/**
+# Copyright (c) Advanced Micro Devices, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the \"License\");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an \"AS IS\" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+**/
+
+// Package health is a collection of utility to access health exporter grpc service 
+// hosted by amd-metrics-exporter service
+package exporter
+
+import (
+    "context"
+    "fmt"
+    "os"
+    "strings"
+    "github.com/ROCm/k8s-device-plugin/internal/pkg/exporter/metricssvc"
+    "google.golang.org/grpc"
+    "google.golang.org/grpc/credentials/insecure"
+    "google.golang.org/protobuf/types/known/emptypb"
+    "github.com/golang/glog"
+    pluginapi "k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1"
+)
+
+const (
+    healthSocket = "/var/lib/amd-metrics-exporter/amdgpu_device_metrics_exporter_grpc.socket"
```

Comment:
This path should come through an environment variable. Its possible that user creates the socket under a different path in the metrics-exporter, that same path would have to get passed to device plugin. So it should not be hardcoded. The socket file name can be hardcoded.